### PR TITLE
Dagger controller tests

### PR DESCRIPTION
**Pull request by jflo**
Demonstrates how to use Dagger to compose BesuControllers for use in privacy tests.

- Separate out flex group privacy test since it requires differently configured dependencies.
- test specific, inline module/component definitions
- reusable module for providing mockable enclaves, besu commands, and genesis

### CODE REVIEW

**Comment by jflo** [comment]
PR passes the `acceptanceTestBftSoak` task.

File: besu/src/test/java/org/hyperledger/besu/PrivacyReorgTest.java [hunk 1]
```diff
@@ -130,11 +141,15 @@ public class PrivacyReorgTest {
           .signAndBuild(KEY_PAIR);
 
   private final BlockDataGenerator gen = new BlockDataGenerator();
-  private BesuController besuController;
-  private PrivateStateRootResolver privateStateRootResolver;
   private PrivacyParameters privacyParameters;
   private Enclave mockEnclave;
   private Transaction privacyMarkerTransaction;
+  private final PrivacyReorgTestComponent context =
```
Comment (fab-10) [inline besu/src/test/java/org/hyperledger/besu/PrivacyReorgTest.java:147]: before the variable was called `component`, should this be renamed for consistency

File: acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/ThreadBesuNodeRunner.java [hunk 2]
```diff
@@ -161,8 +128,12 @@ public void startNode(final BesuNode node) {
...
     }
 
-    final StorageServiceImpl storageService = new StorageServiceImpl();
-    final SecurityModuleServiceImpl securityModuleService = new SecurityModuleServiceImpl();
+    AcceptanceTestBesuComponent component =
+        DaggerThreadBesuNodeRunner_AcceptanceTestBesuComponent.create();
+
+    // final StorageServiceImpl storageService = new StorageServiceImpl();
+    // storageService.registerKeyValueStorage(new
+    // InMemoryStoragePlugin.InMemoryKeyValueStorageFactory("memory"));
```
Comment (fab-10) [inline acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/ThreadBesuNodeRunner.java:136]: there are still few commented code sections, remove if not needed

**Review by fab-10 — COMMENTED** [review]
Overall looks good, just few comments

File: besu/src/main/java/org/hyperledger/besu/services/BesuConfigurationImpl.java [hunk 3]
```diff
@@ -36,6 +37,7 @@ public class BesuConfigurationImpl implements BesuConfiguration {
   private Optional<Integer> rpcHttpPort = Optional.of(8545);
 
   /** Default Constructor. */
+  @Inject
```
[the remark this answers is not in the document]
Comment (jflo) [reply besu/src/main/java/org/hyperledger/besu/services/BesuConfigurationImpl.java]: This is leftover from a prior iteration. It was allowing Dagger to provide these via its default constructor, however that is no longer needed and this has been removed.

File: acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/permissioning/NodeSmartContractPermissioningV2AcceptanceTest.java [hunk 4]
```diff
@@ -20,6 +20,7 @@
 import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 
+@Disabled
```
[the remark this answers is not in the document]
Comment (jflo) [reply acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/permissioning/NodeSmartContractPermissioningV2AcceptanceTest.java]: no good reason, this test has been restored.

**Review by fab-10 — APPROVED** [review]

File: acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/ThreadBesuNodeRunner.java [hunk 5]
```diff
@@ -237,35 +178,20 @@ public void startNode(final BesuNode node) {
             .transactionPoolValidatorService(transactionPoolValidatorServiceImpl)
             .build();
 
+    final BesuControllerBuilder builder = component.besuControllerBuilder();
+    builder.isRevertReasonEnabled(node.isRevertReasonEnabled());
+    builder.networkConfiguration(node.getNetworkingConfiguration());
```
Comment (siladu) [inline acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/ThreadBesuNodeRunner.java:183]: nit: this made me double take - is there a Dagger-y reason not to chain these or just a style thing?

File: acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/ThreadBesuNodeRunner.java [hunk 6]
```diff
@@ -396,4 +321,229 @@ public void startConsoleCapture() {
   public String getConsoleContents() {
     throw new RuntimeException("Console contents can only be captured in process execution");
   }
+
+  @Module
+  @SuppressWarnings("CloseableProvides")
+  static class BesuControllerModule {
```
Comment (siladu) [inline acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/ThreadBesuNodeRunner.java:327]: Wondering about this being nested vs pulled out into its own class...keen to hear your thoughts.

Are we likely to end up with lots of Modules attached to classes or just a few at the top level?

File: besu/src/test/java/org/hyperledger/besu/components/PrivacyParametersModule.java
Comment (siladu) [inline besu/src/test/java/org/hyperledger/besu/components/PrivacyParametersModule.java:1]: What's the thinking behind separate module classes in a components (dagger-specific?) package vs nested alongside the class consuming it?

File: acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/ThreadBesuNodeRunner.java (hunk 5 above)
Comment (jflo) [reply acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/ThreadBesuNodeRunner.java:183]: No, no reason that would be dagger specific.

File: besu/src/test/java/org/hyperledger/besu/components/PrivacyParametersModule.java
Comment (jflo) [reply besu/src/test/java/org/hyperledger/besu/components/PrivacyParametersModule.java:1]: It's the age old package-by-function vs. package-by-feature dilemma. My thinking is that functional changes (what DB we use, configuration subsystem, logging framework) don't require the same level of modularity/reuse as feature based ones. My preference is to package by feature, but that is debatable.

**Pull request merged**